velox/type/tests/DecimalTest.cpp

```diff
@@ -193,5 +193,49 @@ TEST(DecimalTest, valueInPrecisionRange) {
       DecimalUtil::kLongDecimalMin - 1, LongDecimalType::kMaxPrecision));
 }
 
+TEST(DecimalAggregateTest, adjustSumForOverflow) {
+  struct SumWithOverflow {
+    int128_t sum{0};
+    int64_t overflow{0};
+
+    void add(int128_t input) {
+      overflow += DecimalUtil::addWithOverflow(sum, sum, input);
+    }
+
+    std::optional<int128_t> adjustedSum() const {
+      return DecimalUtil::adjustSumForOverflow(sum, overflow);
+    }
+
+    void reset() {
+      sum = 0;
+      overflow = 0;
+    }
+  };
+
+  SumWithOverflow accumulator;
+  // Int128_t.max + 100 will trigger one upward overflow, but the final sum
+  // result calculated by DecimalUtil::addWithOverflow will be negative.
+  accumulator.add(std::numeric_limits<int128_t>::max());
+  accumulator.add(100);
+  accumulator.add(std::numeric_limits<int128_t>::min());
+  EXPECT_EQ(accumulator.adjustedSum(), 99);
```

Comment:
The comment above says "result will be negative" , but the values here is positive (99). This is confusing. Is the comment correct? If so, would it be possible to clarify to avoid confusing future readers?

Reply:
Here, it refers to the fact that the sum before "adjust" is negative. I will fix the comment more explicit.